data_processors/pipeline/domain/workflow.py

```diff
@@ -106,29 +106,29 @@ def get_workdir_root(self):
     def get_output_root(self):
         return libssm.get_ssm_param(self.get_ssm_key_output_root())
 
-    def construct_workdir(self, target_id, timestamp: datetime):
+    def construct_workdir(self, target_id, timestamp: datetime, secondary_target_id:str = None, portal_run_uid: str=None):
         """
         Construct a work directory given target ID and a timestamp
         """
         workdir_root = self._sanitize_gds_path_str(self.get_workdir_root())
-        return workdir_root + "/" + self.get_mid_path(target_id, timestamp)
+        return workdir_root + "/" + self.get_mid_path(target_id, timestamp, secondary_target_id=secondary_target_id, portal_run_uid=portal_run_uid)
 
-    def construct_outdir(self, target_id, timestamp: datetime):
+    def construct_outdir(self, target_id, timestamp: datetime, secondary_target_id:str = None, portal_run_uid: str=None):
```

Comment:
Is there any other reason for allowing `secondary_target_id` and `portal_run_uid` optional; i.e. assigning them `None` here as default. Because this will break in tick next, when we try to construct string path -- _which may be ok if that is the intention_. However, if that the intention and; it seems like it is stronger case the fact that, they both are mandatory fields/parameter now; then let the caller supply it. Be it even an empty string, if caller like. Making interface stronger requirement...

Reply:
> Is there any other reason for allowing secondary_target_id and portal_run_uid optional

`secondary_target_id` is not needed by the `PrimaryWorkflowHelper` class since it can use the run name along with the portal run uid.  

The portal run uid can probably be mandatory.